OMCompiler/SimulationRuntime/c/simulation/solver/spatialDistribution.c

```diff
@@ -365,10 +384,11 @@ double spatialDistribution(DATA* data, threadData_t *threadData, unsigned int in
     realDirection = 0 /* standing still */;
   }
 
-  /* What should happen if realDirection doesn't match given isPositiveVelocity and velocity is not 0? */
+  /* If real direction doesn't match isPositiveVelocity just flip it.
+   * This still indicates something wrong, so we don't extrapolate the output */
   if ((isPositiveVelocity && realDirection==-1 && deltaX > SPATIAL_ZERO_DELTA_X) || (!isPositiveVelocity && realDirection!= -1 && deltaX > SPATIAL_ZERO_DELTA_X)) {
-    errorStreamPrint(LOG_STDOUT, 0, "Boolean isPositiveDirection doesn't match with direction x is moving.");
-    omc_throw_function(threadData);
+    isPositiveVelocity = isPositiveVelocity ? 0:1;
```

Comment:
Same here... (-1,0,1) to (0,1,0). Why?

Reply:
Changed to `isPositiveVelocity  = !isPositiveVelocity;`